src/main/java/com/facebook/openwifirrm/modules/operationelement/HTOperationElement.java

```diff
@@ -0,0 +1,162 @@
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import org.apache.commons.codec.binary.Base64;
+
+import com.facebook.openwifirrm.Utils;
+
+/**
+ * High Throughput (HT) Operation Element, which are potentially present in
+ * wifiscan entries. Details in IEEE 802.11-2020 standard.
+ */
+public class HTOperationElement implements OperationElement {
+
+	protected final byte primaryChannel;
+	protected final byte secondaryChannelOffset;
+	protected final boolean staChannelWidth;
+	protected final boolean rifsMode;
+	protected final byte htProtection;
+	protected final boolean nongreenfieldHtStasPresent;
+	protected final boolean obssNonHtStasPresent;
+	protected final byte channelCenterFrequencySegment2;
+	protected final boolean dualBeacon;
+	protected final boolean dualCtsProtection;
+	protected final boolean stbcBeacon;
+	protected final byte[] basicHtMcsSet;
+
+	/**
+	 * XXX some combinations of these parameters may be invalid as defined by
+	 * 802.11-2020, but this is not checked here. If fidelity to 802.11 is required,
+	 * the caller of this method must make sure to pass in valid parameters. The
+	 * 802.11-2020 specification has more details about the parameters.
+	 *
+	 * @param primaryChannel                 channel index
+	 * @param secondaryChannelOffset
+	 * @param staChannelWidth
+	 * @param rifsMode
+	 * @param htProtection
+	 * @param nongreenfieldHtStasPresent
+	 * @param obssNonHtStasPresent
+	 * @param channelCenterFrequencySegment2
+	 * @param dualBeacon
+	 * @param dualCtsProtection
+	 * @param stbcBeacon
+	 * @return base64 encoded ht operator as a String
+	 */
+	public HTOperationElement(byte primaryChannel, byte secondaryChannelOffset, boolean staChannelWidth,
+			boolean rifsMode, byte htProtection, boolean nongreenfieldHtStasPresent, boolean obssNonHtStasPresent,
+			byte channelCenterFrequencySegment2, boolean dualBeacon, boolean dualCtsProtection, boolean stbcBeacon) {
+		this.primaryChannel = primaryChannel;
+		this.secondaryChannelOffset = secondaryChannelOffset;
+		this.staChannelWidth = staChannelWidth;
+		this.rifsMode = rifsMode;
+		this.htProtection = htProtection;
+		this.nongreenfieldHtStasPresent = nongreenfieldHtStasPresent;
+		this.obssNonHtStasPresent = obssNonHtStasPresent;
+		this.channelCenterFrequencySegment2 = channelCenterFrequencySegment2;
+		this.dualBeacon = dualBeacon;
+		this.dualCtsProtection = dualCtsProtection;
+		this.stbcBeacon = stbcBeacon;
+		// the next 16 bytes are for the basic HT-MCS set
+		// a default is chosen; if needed, we can add a parameter to set these
+		this.basicHtMcsSet = new byte[16];
+	}
+
+	public HTOperationElement() {
+		this((byte) 1, (byte) 0, false, false, (byte) 0, true, false, (byte) 0, false, false, false);
+	}
+
+	/**
+	 *
+	 * @param htOperString must be a String representing a base64 encoded properly
+	 *                     formatted ht operation element (see 802.11-2020 standard)
+	 */
+	public HTOperationElement(String htOperString) {
```

Comment:
NOTE: I have not checked this for correctness (or the VHT version). @pohanhf if you want to check when you're available.

Reply:
@elludraon do you want to wait for Po-Han to check this?

Reply:
Haleema checked it.